epilogue-runtime/src/main/java/edu/wpi/first/epilogue/logging/FileBackend.java

```diff
@@ -185,4 +183,186 @@ public <P, M extends ProtoMessage<M>> void log(String identifier, P value, Proto
 
     ((ProtobufLogEntry<P>) m_entries.get(identifier)).append(value);
   }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public <P, M extends ProtoMessage<M>> void log(
+      String identifier, P value, Protobuf<P, M> proto, LogMetadata metadata) {
+    // DataLog.addSchema has checks that we're able to skip, avoiding allocations
+    if (m_seenProtos.add(proto)) {
+      m_dataLog.addSchema(proto);
+    }
+
+    if (!m_entries.containsKey(identifier)) {
+      var entry = ProtobufLogEntry.create(m_dataLog, identifier, proto);
+      if (!metadata.dependencies().isEmpty()) {
```

Comment:
Let's have a function on `LogMetadata` to check if it's empty or not. For now, it'd just look at the dependencies list, but could be changed in the future if more types of metadata are added